File: src/main/java/ru/afso/projectzero/models/Survey.java
package ru.afso.projectzero.models;

import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;
import java.util.Date;
import java.util.List;

@Document(collection = "surveys")
public class Survey extends BaseModel {

    private Date createdAt;

    private Date beginDate;

    private Date endDate;

    private String title;

    private String description;

    private List<Question> questions;

    @DBRef
    private Research research;

    // Change to geojson or something else
    private Object position;

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Date getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(Date createdAt) {
        this.createdAt = createdAt;
    }

    public Date getBeginDate() {
        return beginDate;
    }

    public void setBeginDate(Date beginDate) {
        this.beginDate = beginDate;
    }

    public Date getEndDate() {
        return endDate;
    }

    public void setEndDate(Date endDate) {
        this.endDate = endDate;
    }

    public List<Question> getQuestions() {
        return questions;
    }

    public void setQuestions(List<Question> questions) {
        this.questions = questions;
    }

    public Research getResearch() {
        return research;
    }

    public void setResearch(Research research) {
        this.research = research;
    }

    public Object getPosition() {
        return position;
    }

    public void setPosition(Object position) {
        this.position = position;
    }
}

File: client/surveyclient/surveyclient/surveyclient/Models/Survey.cs
using System;
using System.Collections.Generic;

namespace surveyclient.Models
{
    public class Survey : BaseModel
    {
        /// <summary>
        /// Дата создания опроса. Для пользователя скрыто
        /// </summary>
        public DateTime CreatedAt { get; set; }

        /// <summary>
        /// Дата начала опроса. Для пользователя скрыто.
        /// Должно быть выставлено при начале опроса
        /// </summary>
        public DateTime? BeginDate { get; set; }

        /// <summary>
        /// Дата окончания опроса. Для пользователя скрыто
        /// Должно быть выставлено при завершении опроса
        /// </summary>
        public DateTime? EndDate { get; set; }

        /// <summary>
        /// Название опроса.
        /// </summary>
        public string Title { get; set; }

        /// <summary>
        /// Описание опроса.
        /// </summary>
        public string Description { get; set; }

        /// <summary>
        /// Список вопросов.
        /// </summary>
        public List<Question> Questions { get; set; }
    }
}

